src/components/atoms/button/stories/button.stories.common.js

```diff
@@ -1,47 +1,132 @@
-import { Text, View, Switch } from 'react-native';
-import { css } from '@emotion/native';
-import { useTheme } from 'emotion-theming';
-import { SafeView } from 'storybook/story-components-native';
-import { ThemeProvider, ThemeActionsContext } from 'store';
+import { withKnobs, text } from '@storybook/addon-knobs';
+import StoryPage, {
+  Description,
+  DocText,
+  DocSection,
+  DocItem,
+  DocCode,
+  DocExternalLink,
+  InlineCode,
+  PropTable,
+  StyleList,
+  TextList,
+  ThemeSwitcher
+} from 'storybook/story-components';
 
 import Button from '../';
 
-const wrapperStyle = css`
-  flex-direction: row;
-  justify-content: center;
-  align-items: center;
-  margin: 10px 0;
-`;
-
-const ThemeSwitcher = () => {
-  const { toggleTheme } = React.useContext(ThemeActionsContext);
-  const theme = useTheme();
-  return (
-    <View style={wrapperStyle}>
-      <Text>Switch theme</Text>
-      <Switch onValueChange={toggleTheme} value={theme.name !== 'light'} testID="theme-switch" />
-    </View>
-  );
-};
-
 storiesOf('Atoms/Button', module)
   .addDecorator(storyFn => (
-    <ThemeProvider>
-      <SafeView>{storyFn()}</SafeView>
-    </ThemeProvider>
+    <StoryPage title="Button component" url="components/atoms/button">
+      <Description>
+        <DocText>An example story for a button</DocText>
+      </Description>
+      {storyFn()}
+    </StoryPage>
   ))
-  .add('With Simple Text', () => (
-    <Button onPress={action('Clicked Button')} title="Hello">
-      <Text>Hello Button</Text>
-    </Button>
+  .addDecorator(withKnobs)
+  .add('With title', () => (
+    <DocSection title="With title">
+      <DocItem
+        name="title"
+        description="The title to be used for the buttons content"
+        typeInfo="string"
+        required
+        example={{
+          render: () => <Button onPress={action('Button Pressed')} title={text('text', 'A button with a title')} />,
+          code: '<Button title="A button with a title" onPress={handleButtonPress}>'
+        }}
+      />
+    </DocSection>
+  ))
+  .add('With onPress', () => (
+    <DocSection title="With onPress">
+      <DocItem
+        name="onPress"
+        description="Callback for button press"
+        typeInfo="func"
+        required
+        example={{
+          render: () => <Button onPress={action('Button Pressed')} title="A button" />,
+          code: '<Button title="A button" onPress={handleButtonPress}>'
+        }}
+      />
+    </DocSection>
   ))
-  .add('Red Color', () => <Button onPress={action('Red Button Clicked')} title="Red Button" color="red"></Button>)
   .add('Themed Button', () => (
-    <>
+    <DocSection title="Themed Button">
       <ThemeSwitcher />
-      <Button title="Themed button" />
-    </>
+      <Button title="Themed button" onPress={action('Button Pressed')} />
+    </DocSection>
+  ))
+  .add('An example external link', () => (
+    <DocSection title="An example external link">
```

Comment:
Same with these. It will reduce the boilerplate in your stories 👍

Reply:
There is not a lot I can do here. The first title is for the navigation. The second is the story title. Also, DocSections are designed to contain multiple DocItems if desired.

I could make it so that if a DocItem is passed a `sectionTitle` then it is wrapped in an `DocSection` automatically. Not very intuitive though

eg:

```
<DocItem
  sectionTitle="An example external link"
  name="title"
  description="The title to be used for the buttons content"
  typeInfo="string"
  required
  example={{
    render: () => <Button onPress={action('Button Pressed')} title={text('text', 'A button with a title')} />,
    code: '<Button title="A button with a title" onPress={handleButtonPress}>'
  }}
/>
```

Reply:
If well documented, it makes the stories much clearer to interpret though 👍  Great idea!